src/sentry/search/events/builder.py

```diff
@@ -1405,3 +1421,106 @@ def resolve_additional_groupby(self) -> List[SelectType]:
+            if not resolve_only:
+                self.distributions.append(resolved_function)
+                # Still add to aggregates so groupby is correct
+                self.aggregates.append(resolved_function)
+            return resolved_function
+        if snql_function.snql_set is not None:
+            resolved_function = snql_function.snql_set(arguments, alias)
+            if not resolve_only:
+                self.sets.append(resolved_function)
+                # Still add to aggregates so groupby is correct
+                self.aggregates.append(resolved_function)
+            return resolved_function
+        if snql_function.snql_counter is not None:
+            resolved_function = snql_function.snql_counter(arguments, alias)
+            if not resolve_only:
+                self.counters.append(resolved_function)
+                # Still add to aggregates so groupby is correct
+                self.aggregates.append(resolved_function)
+            return resolved_function
+        return None
+
+    def _default_filter_converter(self, search_filter: SearchFilter) -> Optional[WhereType]:
+        name = search_filter.key.name
+        operator = search_filter.operator
+        value = search_filter.value.value
+
+        lhs = self.resolve_column(name)
+        # resolve_column will try to resolve this name with indexer, and if its a tag the Column will be tags[1]
+        is_tag = isinstance(lhs, Column) and lhs.subscriptable == "tags"
+        if is_tag:
+            value = indexer.resolve(value)
+
+        # timestamp{,.to_{hour,day}} need a datetime string
+        # last_seen needs an integer
+        if isinstance(value, datetime) and name not in TIMESTAMP_FIELDS:
+            value = int(to_timestamp(value)) * 1000
+
+        if name in TIMESTAMP_FIELDS:
+            if (
+                operator in ["<", "<="]
+                and value < self.params["start"]
+                or operator in [">", ">="]
+                and value > self.params["end"]
+            ):
+                raise InvalidSearchQuery(
+                    "Filter on timestamp is outside of the selected date range."
+                )
+
+        # Tags are never null, but promoted tags are columns and so can be null.
+        # To handle both cases, use `ifNull` to convert to an empty string and
+        # compare so we need to check for empty values.
+        if is_tag:
+            if operator not in ["IN", "NOT IN"] and not isinstance(value, int):
+                sentry_sdk.set_tag("query.lhs", lhs)
+                sentry_sdk.set_tag("query.rhs", value)
+                sentry_sdk.capture_message("Tag value was not found", level="error")
+                raise InvalidSearchQuery("Tag value was not found")
+
+        # TODO(wmak): Need to handle `has` queries, basically check that tags.keys has the value?
+
+        if search_filter.value.is_wildcard():
+            raise IncompatibleMetricsQuery("wildcards not supported")
```

Comment:
Should this check be lifted to the top of this function? If it's a wildcard, there's no need to do any additional checks or hit the indexer to resolve anything.